spec/system/admin/product_import_spec.rb

```diff
@@ -675,14 +680,20 @@
       expect(page).to have_selector '.created-count', text: '2'
       expect(page).not_to have_selector '.updated-count'
 
-      puts "TODO: migrate to v3"
-      Flipper.disable(:admin_style_v3) # disabling BUU for legacy products page
+      default_variant_selector = "tr:has(input[aria-label=Name][value='Carrots'])"
+
       visit spree.admin_products_path
 
-      within "#p_#{Spree::Product.find_by(name: 'Carrots').id}" do
-        expect(page).to have_input "product_name", with: "Carrots"
-        expect(page).to have_select "variant_unit_with_scale", selected: "Weight (lb)"
-        expect(page).to have_content "5" # on_hand
+      carrots = Spree::Product.find_by(name: 'Carrots')
+
+      within "#product_#{carrots.id}" do
```

Comment:
Instead of asserting backend values, we can use `ProductsHelper#row_containing_name` to find the table row containing a field in column labelled "name" with the text:
```suggestion
      within row_containing_name("Carrots") do
```

(You'll need to include `ProductsHelper` at the top of the file)

Reply:
I'm not sure I can make it work with `row_containing_name`, as it finds the first row which contains the name "Carrots" (the product row - `TR[1]`):

```
[14] pry(#<RSpec::ExampleGroups::ProductImport::WhenImportingProductsFromUploadedFile>)> page.find(row_containing_name("Carrots"))
=> #<Capybara::Node::Element tag="tr" path="//HTML[1]/BODY[1]/DIV[1]/DIV[4]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/TURBO-FRAME[1]/DIV[3]/DIV[1]/FORM[1]/TABLE[1]/TBODY[3]/TR[1]">
```

I think we're aiming to look at the second row (i.e., the first variant row - `TR[2]`). The current implementation looks at the whole section, without specifying the row:

```
[8] pry(#<RSpec::ExampleGroups::ProductImport::WhenImportingProductsFromUploadedFile>)> page.find("#product_#{carrots.id}")
=> #<Capybara::Node::Element tag="tbody" path="//HTML[1]/BODY[1]/DIV[1]/DIV[4]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/TURBO-FRAME[1]/DIV[3]/DIV[1]/FORM[1]/TABLE[1]/TBODY[3]">
```

> (You'll need to include ProductsHelper at the top of the file)

I've added it to the base_spec_helper so we don't need to include it; do you think it's better to have it loaded only when needed?

Reply:
> I'm not sure I can make it work with row_containing_name

Oh well, thanks for trying, sorry to have made more work for you.

> I've added it to the base_spec_helper so we don't need to include it; do you think it's better to have it loaded only when needed?

Oh I see. As it's for only a specific part of the system, I personally would have only included it on the relevant tests (I guess I forgot that from a prior review). This is fine too :)